webui/react/src/components/ColumnPickerMenu.tsx

```diff
@@ -114,57 +101,54 @@ const ColumnPickerTab: React.FC<ColumnTabProps> = ({
 
     // If uncheck something pinned, reduce the pinnedColumnsCount
     allFilteredColumnsChecked &&
-      updateSettings({
-        pinnedColumnsCount: newColumns.filter(
-          (col) => columnState.indexOf(col) < settings.pinnedColumnsCount,
-        ).length,
-      });
+      onPinnedColumnsCountChange?.(
+        newColumns?.filter((col) => columnState.indexOf(col) < pinnedColumnsCount).length,
+      );
   }, [
     allFilteredColumnsChecked,
     columnState,
     filteredColumns,
     onVisibleColumnChange,
-    settings.pinnedColumnsCount,
-    updateSettings,
+    pinnedColumnsCount,
+    onPinnedColumnsCountChange,
   ]);
 
   const handleColumnChange = useCallback(
     (event: CheckboxChangeEvent) => {
       const { id, checked } = event.target;
       if (id === undefined) return;
-      const pinnedColumnsCount = settings.pinnedColumnsCount;
-      if (settings.compare) {
+      if (compare) {
         // pin or unpin column
         const newColumns = columnState.filter((c) => c !== id);
         if (checked) {
           newColumns.splice(pinnedColumnsCount, 0, id);
-          updateSettings({ pinnedColumnsCount: Math.max(pinnedColumnsCount + 1, 0) });
+          onPinnedColumnsCountChange?.(Math.max(pinnedColumnsCount + 1, 0));
```

Comment:
qq: unpinned columns become pinned column after user remove the pinned columns. is it intetentional? 

https://github.com/determined-ai/determined/assets/109113616/0f845435-7fd3-4bbe-8f3d-6faf06ba1505

Reply:
adding fix